### PR TITLE
Custom Demo Gen Functions

### LINKED ISSUE #653: demo_gen: Add support for custom demonstration functions
Don't have time to do this, but here's what I'd do:

Add to `DemoInfo` in `core/hir/attrs.rs`, specifically a `custom_func()` attribute, where it: 

1. Points to a Javascript file.
2. Lists all of the primitive inputs that the function takes.

Then under `demo_gen/mod.rs`, for every function you find with that attribute, copy the Javascript file to the output folder. Then add the function and parameter information to `RenderInfo`

This would be really nice to set up the Segmenter in ICU4X.

### PR DESCRIPTION
**Pull request by ambiguousname**
The ultimate goal is to get the Segmenter for the ICU4X demo working.

Resolves #653.

### CODE REVIEW

**Comment by ambiguousname** [comment]
Whenever this looks good, my TODOs will be:

1. Updating docs in both this repo and the Demo gen book
2. Updating ICU4X demo to use Segmenter

**Comment by ambiguousname** [comment]
Not ready to merge yet, just realized I have to update the custom function's definition and add a test.

**Comment by ambiguousname** [comment]
*now* this is fully functional and ready to merge after review

File: core/src/hir/attrs.rs [hunk 1]
```diff
@@ -81,6 +81,9 @@ pub struct DemoInfo {
     /// `#[diplomat::demo(external)]` represents an item that we will not evaluate, and should be passed to the rendering engine to provide.
     pub external: bool,
 
+    /// `#[diplomat::demo(custom_func = "/file/name/here.mjs")]` represents a user defined function that they define themselves in JS.
```
Comment (Manishearth) [inline core/src/hir/attrs.rs:84]: could you document more on what this does? I'm not really understanding the effect of setting this right now

**Comment by ambiguousname** [comment]
New associated PR: https://github.com/rust-diplomat/book/pull/9

**Review by Manishearth — APPROVED** [review]

**Pull request merged**